turbinia/workers/bulk_extractor.py

```diff
@@ -37,21 +41,86 @@ def run(self, evidence, result):
     # generate a summary report to stdout so that it could be used for
     # a report in this task.
     # Create the new Evidence object that will be generated by this Task.
-    output_evidence = CompressedDirectory()
+    output_evidence = BulkExtractorOutput()
     # Create a path that we can write the new file to.
     base_name = os.path.basename(evidence.local_path)
     output_file_path = os.path.join(self.output_dir, base_name)
     # Add the output path to the evidence so we can automatically save it
     # later.
-    output_evidence.source = output_file_path
+    output_evidence.local_path = output_file_path
+
+    try:
+      # Generate the command we want to run then execute.
+      cmd = 'bulk_extractor {0:s} -o {1:s}'.format(
+          evidence.local_path, output_file_path)
+      result.log('Running Bulk Extractor as [{0:s}]'.format(cmd))
+      self.execute(cmd, result, new_evidence=[output_evidence], shell=True)
 
-    # Generate the command we want to run.
-    cmd = 'bulk_extractor {0:s} -o {1:s}'.format(
-        evidence.local_path, output_file_path)
-    # Add a log line to the result that will be returned.
-    result.log('Running Bulk Extractor as [{0:s}]'.format(cmd))
-    # Actually execute the binary
-    self.execute(
-        cmd, result, new_evidence=[output_evidence], close=True, shell=True)
+      # Generate bulk extractor report
+      (report, summary) = self.generate_summary_report(output_file_path)
+      output_evidence.text_data = '\n'.join(report)
+      result.report_data = output_evidence.text_data
 
+      # Compress the bulk extractor output directory.
+      output_evidence.compress()
+      result.close(self, success=True, status=summary)
+    except TurbiniaException as exception:
+      result.close(self, success=False, status=str(exception))
+      return result
     return result
+
+  def generate_summary_report(self, output_file_path):
+    """Generate a summary report from the resulting bulk extractor run.
+
+    Args:
+      output_file_path(str): the path to the bulk extractor output.
+
+    Returns:
+      list: formatted report output.
+      str: summary of bulk extractor run.
+    """
+    report = []
+    report_path = os.path.join(output_file_path, 'report.xml')
+
+    # Check if report.xml was not generated by bulk extractor.
+    if not os.path.exists(report_path):
+      report = 'Execution successful, but the report is not available.'
+      return (report, report)
+
+    # Parse existing XML file.
+    xml = xml_tree.parse(report_path)
+
+    # Retrieve summary related results.
+    report.append(fmt.heading1('Bulk Extractor Results'))
```

Comment:
I know it's not super obvious, but you should start the report at heading4 so that it doesn't conflict with other headings that get set as the report is compiled (this is just one sub-section of the overall report).  See here for a bit more info:
https://github.com/google/turbinia/blob/master/docs/developing-new-tasks.md#reporting

If you have ideas as to how to make this more obvious for developers, let me know.  Thanks!